File: Greedy/860. Lemonade Change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int five = 0;
        int ten = 0;
        for (auto b : bills){
            if (b == 5){
                five++;
            }
            else if (b == 10 && five > 0){
                ten++;
                five--;
            }
            else if (b == 20 && ten > 0 && five > 0){
                ten--;
                five--;
            }
            else if(b == 20 && five > 2){
                five -= 3;
            }
            else{
                return false;
            }
        }
        return true;
    }
};

File: Greedy/860. Lemonade Change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        five = 0
        ten = 0
        for b in bills:
            if b == 5:
                five += 1
            elif b == 10 and five > 0:
                ten += 1
                five -= 1
            elif b == 20 and ten > 0 and five > 0:
                ten -= 1
                five -= 1
            elif b == 20 and five > 2:
                five -= 3
            else:
                return 0
        return 1